.github/workflows/ci-repo.yml

```diff
@@ -154,23 +166,33 @@ jobs:
     name: Release
     runs-on: ubuntu-latest
     needs: package
-    if: github.ref == 'refs/heads/main'
+    if: 
+      github.ref == 'refs/heads/main' &&
+      (github.event.inputs.shouldRelease == 'true' || github.event.inputs.shouldRelease == true)
 
     steps:
       - name: Checkout code
         uses: actions/checkout@v4
+      
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: "vscode/.nvmrc"
 
       - name: Download VSIX artifacts
         uses: actions/download-artifact@v3
         with:
           path: ./artifacts
 
-      - name: Generate tag name
+      - name: Read version from package.json
+        id: get_version
+        run: echo "version=$(node -p \"require('./package.json').version\")" >> $GITHUB_OUTPUT
```

Comment:
Do you need to set the working directory?

Reply:
good catch. I definitely need to since we are using the pattern from continue